pandas/core/dtypes/cast.py

```diff
@@ -1094,14 +1093,12 @@ def astype_nansafe(
         The dtype was a datetime64/timedelta64 dtype, but it had no unit.
     """
     if arr.ndim > 1:
-        # Make sure we are doing non-copy ravel and reshape.
-        flags = arr.flags
-        flat = arr.ravel("K")
+        # TODO: try to use contiguity to avoid potentially copying here, see #42475
+        flat = arr.ravel()
         result = astype_nansafe(flat, dtype, copy=copy, skipna=skipna)
-        order: Literal["C", "F"] = "F" if flags.f_contiguous else "C"
         # error: Item "ExtensionArray" of "Union[ExtensionArray, ndarray]" has no
         # attribute "reshape"
-        return result.reshape(arr.shape, order=order)  # type: ignore[union-attr]
+        return result.reshape(arr.shape)  # type: ignore[union-attr]
```

Comment:
if we revert all of #38507, do we remove the need for an ignore? i.e. is there another issue lurking here.

Reply:
Reverting would remove the need for the ignore, in the previous logic the reshape was inside an `isinstance(ndarray)`. 

AFAICT there is no issue here though - the problematic case here would be if `astype_nansafe` is called on a 2D ndarray and we try to convert to an `ExtensionDtype`. But based on https://github.com/pandas-dev/pandas/blob/87d785599e01ec1f8a9799fc87d6ab4c77c3590c/pandas/core/generic.py#L5731-L5738 that should be impossible. So I think we could get rid of the ignore with an assert if desired

Reply:
sure. but let's leave that out of this PR which is being backported